src/net_processing.cpp

```diff
@@ -3396,6 +3408,14 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
 
     if (msg_type == NetMsgType::CMPCTBLOCK)
     {
+        // Stop processing the compact block if there is no mempool
+        if (!m_mempool)
+        {
+            LogPrint(BCLog::NET, "No mempool to reconstruct from compact block sent by peer=%d\n", pfrom.GetId());
+            pfrom.fDisconnect = true;
```

Comment:
This will require to add an early return to `MaybeSetPeerAsAnnouncingHeaderAndIDs`, no?

Otherwise you'd disconnect the best peer(s) after IBD.

Reply:
Removed the action of disconnecting a node to conform to BIP152. Added `m_mempool` checks in `MaybeSetPeerAsAnnouncingHeaderAndIDs` to make sure that a node without a mempool would never request for compact blocks from its peers in the first place.